### PR TITLE
Quarkus CLI 2.0

### PR DESCRIPTION
**Pull request by patriot1burke**
WIP: I'm still working on testing.

This mostly a port of the aesh/ stuff to Picocli.  I did change command structure a bit here and there.  One specific one is that "list" is renamed to "extensions".  I want the "list" command to list extensions and dependencies that were added to the project with the "add" command.

I'll add some additional comments to this PR that show usage.

### CODE REVIEW

**Comment by patriot1burke** [comment]
<pre>Usage: <b>quarkus</b> [<font color="#AA5500">-ehV</font>] [COMMAND]
  <font color="#AA5500">-e</font>, <font color="#AA5500">--errors</font>    Produce execution error messages.
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>      Show this help message and exit.
  <font color="#AA5500">-V</font>, <font color="#AA5500">--version</font>   Print version information and exit.
Commands:
  <b>build</b>             Build your quarkus project
  <b>clean</b>             Clean current project
  <b>create</b>            Create a new quarkus project.
  <b>add</b>               Add extension(s) to current project.
  <b>remove</b>, <b>rm</b>        Remove an extension from this project.
  <b>dev</b>               Execute project in live coding dev mode
  <b>extensions</b>, <b>exts</b>  Display a list of available Quarkus extensions</pre>

**Comment by patriot1burke** [comment]
<pre>Usage: <b>quarkus create</b> [<font color="#AA5500">-0hV</font>] [<font color="#AA5500">-a</font>=<i>AID</i>] [<font color="#AA5500">-g</font>=<i>GID</i>] [<font color="#AA5500">-v</font>=<i>TAG</i>] [<font color="#AA5500">--maven</font> | <font color="#AA5500">--gradle</font> |
                      <font color="#AA5500">--grade-kotlin-dsl</font>] [<font color="#AA5500">--java</font> | <font color="#AA5500">--kotlin</font> | <font color="#AA5500">--scala</font>]
                      [<font color="#AA5500">EXTENSION</font>...]
Create a new quarkus project.
      [<font color="#AA5500">EXTENSION</font>...]       extension to add to project
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>               Show this help message and exit.
  <font color="#AA5500">-V</font>, <font color="#AA5500">--version</font>            Print version information and exit.
  <font color="#AA5500">-g</font>, <font color="#AA5500">--groupId</font>=<i>GID</i>        The groupId for project
  <font color="#AA5500">-a</font>, <font color="#AA5500">--artifactId</font>=<i>AID</i>     The artifactId for project
  <font color="#AA5500">-v</font>, <font color="#AA5500">--proj-version</font>=<i>TAG</i>   The version for project
  <font color="#AA5500">-0</font>, <font color="#AA5500">--skipExample</font>        Generate without example code.
<font color="#AA5500"> </font>     <font color="#AA5500">--maven</font>              Create a Maven project. (default)
<font color="#AA5500"> </font>     <font color="#AA5500">--gradle</font>             Create a Gradle project.
<font color="#AA5500"> </font>     <font color="#AA5500">--grade-kotlin-dsl</font>   Create a Gradle Kotlin DSL project.
<font color="#AA5500"> </font>     <font color="#AA5500">--java</font>               Generate Java examples. (default)
<font color="#AA5500"> </font>     <font color="#AA5500">--kotlin</font>             Generate Kotlin examples.
<font color="#AA5500"> </font>     <font color="#AA5500">--scala</font>              Generate Scala examples.
</pre>

**Comment by patriot1burke** [comment]
@maxandersen Not sure who else to add to review.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (gastaldi) [inline devtools/cli/src/main/java/io/quarkus/cli/Create.java:99]: Looks like brackets are no longer a thing in 2020 :)

Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:99]: Seriously?

Comment (gastaldi) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:99]: Well, yeah :) IMHO it looks better with brackets :)

**Comment by gsmet** [comment]
Minor nitpicks: I think you should be consistent for the long options casing:
- `groupId` -> `group-id`
- `artifactId` -> `artifact-id`
- `proj-version` -> `project-version`
- `skipExample` -> `skip-example` (I would even prefer `skip-examples` as I think you might have several) @ia3andy I think we should change it in the code starts code if I'm right. Plural is more natural in English.

I would drop the current `--version` from `create` and replace `--project-version` with it.

`quarkus -v` displaying the version makes sense.
`quarkus create -v` displaying the version makes less sense IMHO.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (gastaldi) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:99]: It's all fun and games until someone adds a new line and wonders why it doesn't work as it should :wink:

**Comment by gastaldi** [comment]
I think `aesh` can be removed from the `bom/application/pom.xml` too?

**Comment by patriot1burke** [comment]
> Minor nitpicks: I think you should be consistent for the long options casing:
> 
> * `groupId` -> `group-id`
> * `artifactId` -> `artifact-id`
> * `proj-version` -> `project-version`
> * `skipExample` -> `skip-example` (I would even prefer `skip-examples` as I think you might have several) @ia3andy I think we should change it in the code starts code if I'm right. Plural is more natural in English.
> 
> I would drop the current `--version` from `create` and replace `--project-version` with it.
> 
> `quarkus -v` displaying the version makes sense.
> `quarkus create -v` displaying the version makes less sense IMHO.

This is gonna sound stupid, but i renamed from project-version to proj-version for Usage formatting reasons so I could keep the 80 char width.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:99]: I just don't agree. :) We're not coding in 'vi' or 'ed' anymore.

**Comment by gsmet** [comment]
> This is gonna sound stupid, but i renamed from project-version to proj-version for Usage formatting reasons so I could keep the 80 char width.

An even better reason to use `--version` (if possible).

**Comment by patriot1burke** [comment]
> Minor nitpicks: I think you should be consistent for the long options casing:
> 
> * `groupId` -> `group-id`
> * `artifactId` -> `artifact-id`
> * `proj-version` -> `project-version`
> * `skipExample` -> `skip-example` (I would even prefer `skip-examples` as I think you might have several) @ia3andy I think we should change it in the code starts code if I'm right. Plural is more natural in English.
> 
> I would drop the current `--version` from `create` and replace `--project-version` with it.
> 
> `quarkus -v` displaying the version makes sense.
> `quarkus create -v` displaying the version makes less sense IMHO.

I made the flags consistent with quarkus:XXXX where it made sense.  Either we're keeping the flags consistent  or we aren't.  CHOOSE!

**Comment by gsmet** [comment]
Yeah, the `skipExample` one, I think we should change in quarkus:xxx too. But we need to check with @ia3andy if it makes sense.

As for `version`, I think it's just that it's probably something we couldn't use with Maven. Here it makes perfect sense to use it if we can.

If we can make things better with this CLI, let's make them, I think the idea is to have people use it rather than the Maven counterpart.

**Comment by patriot1burke** [comment]
> I made the flags consistent with quarkus:XXXX where it made sense. Either we're keeping the flags consistent or we aren't. CHOOSE!

For this I meant whether kebel case should be used or not "artifactId" vs. "artifact-id".

**Comment by gastaldi** [comment]
+1 to kebab case everywhere. That's the same pattern used in the JSON files too

**Comment by patriot1burke** [comment]
<pre>Usage: <b>quarkus list</b> [<font color="#AA5500">-hi</font>] [<font color="#AA5500">-s</font>=<i>PATTERN</i>] [<font color="#AA5500">--name</font> | <font color="#AA5500">--concise</font> | <font color="#AA5500">--full</font>]
List installed (default) or installable extensions.
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>             display this help message
  <font color="#AA5500">-i</font>, <font color="#AA5500">--installable</font>      Display installable extensions.
  <font color="#AA5500">-s</font>, <font color="#AA5500">--search</font>=<i>PATTERN</i>   Search filter on extension list. The format is based
                           on Java Pattern.
<font color="#AA5500"> </font>     <font color="#AA5500">--name</font>             Display extension name only. (default)
<font color="#AA5500"> </font>     <font color="#AA5500">--concise</font>          Display extension name and description.
<font color="#AA5500"> </font>     <font color="#AA5500">--full</font>             Display concise format and version related columns.
</pre>

**Comment by patriot1burke** [comment]
Ok, see updated usage messages above base, create, and list.

I removed "extensions" command and replaced it back to "list".  By default "list" will list installed extensions.  "--installable" will list installable extensions.  I also removed descriptor text like "Current extensions blah blah blah" as you might want to pipe this stuff in a script.

**Comment by ia3andy** [comment]
Regarding `skip-examples`, I will change it in codestarts/maven too, but I would personally prefer just: `—empty` as we discussed on quarkus-dev.

**Comment by patriot1burke** [comment]
@ia3andy changed to --empty

Added tests.  Everything should be good to go.

File: devtools/cli/src/main/java/io/quarkus/cli/Add.java
Comment (ia3andy) [inline devtools/cli/src/main/java/io/quarkus/cli/Add.java:34]: @aloubyansky @maxandersen @patriot1burke , this command is not going to be compatible with gradle. This is why for the add/remove/list extension, I would delegate the call to the wrapper plugin to make this command compatible on both mvn and gradle. If there is no wrapper, then you could fallback to this and if it target a gradle project fail with instruction.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (ia3andy) [inline devtools/cli/src/main/java/io/quarkus/cli/Create.java:109]: @patriot1burke fyi `sourceType` is not used by codestarts, you could remove it. What you will need to do though, is to add the language extension when `--kotlin` or `--scala` is selected.

File: devtools/cli/src/main/java/io/quarkus/cli/Add.java
Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Add.java:34]: And even more now that I see you already implemented an `ExecuteUtil` :)

**Review by ia3andy — CHANGES_REQUESTED** [review]
Very nice! Just a comment about add/remove/list commands..

File: devtools/cli/src/main/java/io/quarkus/cli/Add.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Add.java:34]: Can you explain why it won't work?

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:109]: Don't know what "add the language extension" means...

Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:109]: @patriot1burke:
- java -> default
- kotlin -> extension `quarkus-kotlin`
- scala -> extension `quarkus-scala`

in the end the `--kotlin` or `--scala` are just shortcuts.

File: devtools/cli/src/main/java/io/quarkus/cli/Add.java
Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Add.java:34]: sure @patriot1burke, because Gradle API is not available outside of a gradle plugin, without it, we are not able to properly list extensions (neither add and remove which rely on the list): https://github.com/quarkusio/quarkus/blob/master/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/GenericGradleBuildFile.java

**Comment by maxandersen** [comment]
First off - maan this is awesome to have kicked off; it feels so much more natural and complete :) ...will be adding some comments for some issues I found and comments!

File: devtools/cli/src/main/java/io/quarkus/cli/Remove.java
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/Remove.java:42]: `quarkus rm hibernate` in a project with just `quarkus-hibernate-orm-panache` still complains about having to be precise:

```
❌ Multiple extensions matching 'hibernate'
     * io.quarkus:quarkus-hibernate-orm
     * io.quarkus:quarkus-hibernate-reactive
     * io.quarkus:quarkus-hibernate-envers
     * io.quarkus:quarkus-hibernate-orm-panache
     * io.quarkus:quarkus-hibernate-orm-rest-data-panache
     * io.quarkus:quarkus-hibernate-orm-panache-kotlin
     * io.quarkus:quarkus-hibernate-search-elasticsearch
     * io.quarkus:quarkus-hibernate-validator
     Be more specific e.g using the exact name or the full GAV.
```

File: devtools/cli/src/main/java/io/quarkus/cli/core/ExecuteUtil.java
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/core/ExecuteUtil.java:64]: when no no `mvnw` nor `mvn` in path this seem to just fail silently. no error output, no nothing.

File: devtools/cli/pom.xml
Comment (maxandersen) [inline devtools/cli/pom.xml:57]: Tried building with `-Dnative` and I get errors; at least on OSX.

```
Error: Unsupported features in 3 methods
Detailed message:
Error: Class initialization of io.netty.util.internal.logging.Log4JLogger failed. Use the option --initialize-at-run-time=io.netty.util.internal.logging.Log4JLogger to explicitly request delayed initialization of this class.
Original exception that caused the problem: java.lang.NoClassDefFoundError: org/apache/log4j/Priority
```

File: devtools/cli/src/main/java/io/quarkus/cli/Add.java
Comment (maxandersen) [reply devtools/cli/src/main/java/io/quarkus/cli/Add.java:34]: can confirm it doesn't work:

```
 quarkus add hibernate-orm-panache
Unable to add extension: This feature is not yet implemented outside of the Gradle Plugin.
```

File: devtools/cli/src/main/java/io/quarkus/cli/Build.java
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/Build.java:14]: all this parent/spec code and possibly also the error handling we could move to a base class and simplify these commands quite a bit.

File: devtools/cli/src/main/java/io/quarkus/cli/Build.java
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/Build.java:10]: should we have a `--verbose` that would print out the commands actually executed ?

File: devtools/cli/src/main/java/io/quarkus/cli/Build.java
Comment (maxandersen) [inline devtools/cli/src/main/java/io/quarkus/cli/Build.java:8]: looks like I can do `quarkus clean build` and that gets execute as multiple commands. That makes it slower than corresponding `mvn clean build` or `gradle clean build`

Wondering if we could "group" these rather than be done separately ?

**Review by maxandersen — CHANGES_REQUESTED** [review]
overall this looks awesome IMO. 

My comments are probably for most part things we can do in follow up PR's; only "glitch" I think should be fixed before merging is that "add"/"remove" aren't working on gradle projects.

File: devtools/cli/src/main/java/io/quarkus/cli/Add.java
Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Add.java:68]: :-)

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:100]: yes!

File: devtools/cli/src/main/java/io/quarkus/cli/Build.java
Comment (maxandersen) [reply devtools/cli/src/main/java/io/quarkus/cli/Build.java:7]: haven't played with it but picocli lets you customize the execution strategy. https://picocli.info/#_executing_commands_with_subcommands

maybe that could work ?

File: devtools/cli/src/main/java/io/quarkus/cli/core/ExecuteUtil.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/core/ExecuteUtil.java]: stderr wasn't being outputted.  All this code has changed and this is fixed.

File: devtools/cli/pom.xml
Comment (patriot1burke) [reply devtools/cli/pom.xml:57]: added --native, -n switch.

File: devtools/cli/src/main/java/io/quarkus/cli/Build.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Build.java:7]: creating a strategy was the trick.

File: devtools/cli/src/main/java/io/quarkus/cli/Remove.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Remove.java:48]: This should be logged as a separate issue.  Not a CLI thing.

File: devtools/cli/src/main/java/io/quarkus/cli/Build.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Build.java:12]: I used a strategy.  That worked.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:1]: How is that simpler than:
$ quarkus create resteasy

???

File: devtools/cli/pom.xml
Comment (patriot1burke) [reply devtools/cli/pom.xml:57]: added --native switch to build

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:1]: Create a separate issue if you want this please.

**Comment by patriot1burke** [comment]
<pre>Usage: <b>quarkus</b> [<font color="#AA5500">-ehV</font>] [<font color="#AA5500">--verbose</font>] [COMMAND]
  <font color="#AA5500">-e</font>, <font color="#AA5500">--errors</font>    Produce execution error messages.
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>      Show this help message and exit.
  <font color="#AA5500">-V</font>, <font color="#AA5500">--version</font>   Print version information and exit.
      <font color="#AA5500">--verbose</font>   Verbose mode.
Commands:
  <b>build</b>       Build your quarkus project
  <b>clean</b>       Clean current project
  <b>create</b>      Create a new quarkus project.
  <b>list</b>        List installed (default) or installable extensions.
  <b>add</b>         Add extension(s) to current project.
  <b>remove</b>, <b>rm</b>  Remove an extension from this project.
  <b>dev</b>         Execute project in live coding dev mode
</pre>

**Comment by patriot1burke** [comment]
<pre>Usage: <b>quarkus build</b> [<font color="#AA5500">-hns</font>] [<font color="#AA5500">--offline</font>]
Build your quarkus project
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>         display this help message
  <font color="#AA5500">-n</font>, <font color="#AA5500">--native</font>       Build native executable.
      <font color="#AA5500">--offline</font>      Work offline.
  <font color="#AA5500">-s</font>, <font color="#AA5500">--skip-tests</font>   Skip tests.
</pre>

**Comment by patriot1burke** [comment]
<pre>Usage: <b>quarkus create</b> [<font color="#AA5500">-0h</font>] [<font color="#AA5500">-a</font>=<i>ARTIFACT-ID</i>] [<font color="#AA5500">-g</font>=<i>GROUP-ID</i>] [<font color="#AA5500">-v</font>=<i>VERSION</i>] [<font color="#AA5500">--maven</font> | <font color="#AA5500">--gradle</font> |
                      <font color="#AA5500">--grade-kotlin-dsl</font>] [<font color="#AA5500">--java</font> | <font color="#AA5500">--kotlin</font> | <font color="#AA5500">--scala</font>] [<font color="#AA5500">EXTENSION</font>...]
Create a new quarkus project.
      [<font color="#AA5500">EXTENSION</font>...]              extension to add to project
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>                      display this help message
  <font color="#AA5500">-g</font>, <font color="#AA5500">--group-id</font>=<i>GROUP-ID</i>         The groupId for project
  <font color="#AA5500">-a</font>, <font color="#AA5500">--artifact-id</font>=<i>ARTIFACT-ID</i>   The artifactId for project
  <font color="#AA5500">-v</font>, <font color="#AA5500">--version</font>=<i>VERSION</i>           The version for project
  <font color="#AA5500">-0</font>, <font color="#AA5500">--empty</font>                     Generate without example code.
<font color="#AA5500"> </font>     <font color="#AA5500">--maven</font>                     Create a Maven project. (default)
<font color="#AA5500"> </font>     <font color="#AA5500">--gradle</font>                    Create a Gradle project.
<font color="#AA5500"> </font>     <font color="#AA5500">--grade-kotlin-dsl</font>          Create a Gradle Kotlin DSL project.
<font color="#AA5500"> </font>     <font color="#AA5500">--java</font>                      Generate Java examples. (default)
<font color="#AA5500"> </font>     <font color="#AA5500">--kotlin</font>                    Generate Kotlin examples.
<font color="#AA5500"> </font>     <font color="#AA5500">--scala</font>                     Generate Scala examples.
</pre>

**Comment by patriot1burke** [comment]
<pre>List installed (default) or installable extensions.
  <font color="#AA5500">-h</font>, <font color="#AA5500">--help</font>             display this help message
  <font color="#AA5500">-i</font>, <font color="#AA5500">--installable</font>      Display installable extensions.
  <font color="#AA5500">-s</font>, <font color="#AA5500">--search</font>=<i>PATTERN</i>   Search filter on extension list. The format is based on Java Pattern.
<font color="#AA5500"> </font>     <font color="#AA5500">--name</font>             Display extension name only. (default)
<font color="#AA5500"> </font>     <font color="#AA5500">--concise</font>          Display extension name and description.
<font color="#AA5500"> </font>     <font color="#AA5500">--full</font>             Display concise format and version related columns.
</pre>

**Comment by patriot1burke** [comment]
@maxandersen I think I addressed all issues and added some more functionality.  See above for updated usage.  Assuming I can get CI to pass, this should be good to go.  That's a big IF though...I've been unable to get CI to pass this whole last week even though I can get it to pass locally on windows and linux.

**Comment by patriot1burke** [comment]
@maxandersen Yay a passed build.  If you think everything is cool an approval would be welcome.

**Comment by patriot1burke** [comment]
@maxandersen native is going to have to come with a different PR.

**Comment by maxandersen** [comment]
Awesome - will look when back over weekend or first Monday!

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (maxandersen) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:1]: create takes extensions as arguments not a codestart thus you can end up with a mix of things in your project. Which is good when exploring; less great when trying to write docs or follow a guide that just want "take this starting point, add these extensions and we'll walk you through the rest".

i'll open seperate issue.

**Comment by maxandersen** [comment]
I think this is good to go. still things needed but as we aren't shipping any bits yet using this its not hurting to go in and we can get started on the bigger things.

LGTM!

**Review by maxandersen — APPROVED** [review]
LGTM!

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (ia3andy) [inline devtools/cli/src/main/java/io/quarkus/cli/Create.java:85]: you could do it only once I guess

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (ia3andy) [inline devtools/cli/src/main/java/io/quarkus/cli/Create.java:32]: @maxandersen didn't we say we should use `no-examples` in the end?

Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:32]: Fine to change this, but I'm gonna kill somebody if I have to spend another week trying to get CI to pass over such a small change.

**Comment by patriot1burke** [comment]
@maxandersen @ia3andy FYI: Maybe i just don't have gradle set up correctly, but any command (build, clean, list, add, remove) that goes through gradle is painfully slow.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (ia3andy) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:32]: @patriot1burke I can understand this 😂, it shouldn't break CI though :) let's wait for @maxandersen confirmation first.. 

This is not "that" small because it actually changes the spec of the CLI.. ex: if you give an example in a blog post and it changes in the next release, it's not great 😅

Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:32]: Well, you guys have already changed this flag 3 times in the last two weeks and , well, i don't like any of your changes :)  So, if you change your mind yet again in the next two weeks, feel free to do the change yourself.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (patriot1burke) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java:85]: it saves an object creation when its --java and no extensions.

File: devtools/cli/src/main/java/io/quarkus/cli/Create.java
Comment (maxandersen) [reply devtools/cli/src/main/java/io/quarkus/cli/Create.java]: i wasn't aware we were changing them here as those things were stil fluid.

lets just get this in and we can clean up / align these rather than do another round.

**Comment by maxandersen** [comment]
lets get this into master - it just missed the 1.8 cut off ;/

**Review by ia3andy — APPROVED** [review]
LGTM

**Pull request merged**
